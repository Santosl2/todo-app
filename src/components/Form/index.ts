export * from "./Input";
export * from "./MainForm";
export * from "./Button";
