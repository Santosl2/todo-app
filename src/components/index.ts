export * from "./Header";
export * from "./Form";
export * from "./Badge";
export * from "./TextWithBadge";
export * from "./Item";
export * from "./NoResults";
