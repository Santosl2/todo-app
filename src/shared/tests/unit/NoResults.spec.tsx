import { NoResults } from "@/components";
import { render, screen } from "@testing-library/react";

describe("Badge", () => {
  it("should render correctly", () => {
    render(<NoResults />);

    expect(
      screen.getByText("Você ainda não tem tarefas cadastradas")
    ).toBeInTheDocument();
    expect(
      screen.getByText("Crie tarefas e organize seus itens a fazer")
    ).toBeInTheDocument();
    expect(screen.getByText(/Feito por/)).toBeInTheDocument();
  });

  it("match snapshot", () => {
    const { container } = render(<NoResults />);

    expect(container).toMatchSnapshot();
  });
});
