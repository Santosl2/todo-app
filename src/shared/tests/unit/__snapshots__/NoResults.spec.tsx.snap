// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge match snapshot 1`] = `
<div>
  <div
    class="flex h-60 flex-col items-center justify-center rounded-lg border-t border-gray-400 text-gray-300"
  >
    <svg
      fill="currentColor"
      height="56"
      viewBox="0 0 256 256"
      width="56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="256"
        width="256"
      />
      <line
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
        x1="96"
        x2="160"
        y1="128"
        y2="128"
      />
      <line
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
        x1="96"
        x2="160"
        y1="160"
        y2="160"
      />
      <path
        d="M56,40H200a8,8,0,0,1,8,8V200a24,24,0,0,1-24,24H72a24,24,0,0,1-24-24V48A8,8,0,0,1,56,40Z"
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
      />
      <line
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
        x1="80"
        x2="80"
        y1="24"
        y2="56"
      />
      <line
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
        x1="128"
        x2="128"
        y1="24"
        y2="56"
      />
      <line
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
        x1="176"
        x2="176"
        y1="24"
        y2="56"
      />
    </svg>
     
    <strong
      class="mt-4"
    >
      Você ainda não tem tarefas cadastradas
    </strong>
    <p>
      Crie tarefas e organize seus itens a fazer
    </p>
    <small>
      Feito por
       
      <a
        class="underline text-cyan-500 underline-offset-2"
        href="https://github.com/Santosl2"
      >
        Matheus Filype
      </a>
    </small>
  </div>
</div>
`;
